audio_helper/src/openxtpackets.h

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright (c) 2015 Assured Information Security, Inc
+//
+// Dates Modified:
+//  - 4/8/2015: Initial commit
+//    Rian Quinn <quinnr@ainfosec.com>
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef OPENXT_PACKETS_H
+#define OPENXT_PACKETS_H
+
+#include "openxtsettings.h"
+
+typedef enum PacketOpCode {
+
+    // Global
+    OPENXT_FINI                         = 01,
+
+    // Init Commands
+    OPENXT_PLAYBACK_INIT                = 10,
+    OPENXT_PLAYBACK_INIT_ACK            = 11,
+    OPENXT_CAPTURE_INIT                 = 12,
+    OPENXT_CAPTURE_INIT_ACK             = 13,
+
+    // Fini Commands
+    OPENXT_PLAYBACK_FINI                = 20,
+    OPENXT_CAPTURE_FINI                 = 21,
+
+    // Available
+    OPENXT_PLAYBACK_GET_AVAILABLE       = 30,
+    OPENXT_PLAYBACK_GET_AVAILABLE_ACK   = 31,
+    OPENXT_CAPTURE_GET_AVAILABLE        = 32,
+    OPENXT_CAPTURE_GET_AVAILABLE_ACK    = 33,
+
+    // Control
+    OPENXT_PLAYBACK_ENABLE_VOICE        = 40,
+    OPENXT_PLAYBACK_DISABLE_VOICE       = 41,
+    OPENXT_PLAYBACK_SET_VOLUME          = 43,
+    OPENXT_CAPTURE_ENABLE_VOICE         = 44,
+    OPENXT_CAPTURE_DISABLE_VOICE        = 45,
+
+    // Process
+    OPENXT_PLAYBACK                     = 50,
+    OPENXT_CAPTURE                      = 51,
+    OPENXT_CAPTURE_ACK                  = 53,
+
+} PacketOpCode;
+
+typedef struct OpenBlankPacket {
```

Comment:
in response to c)  that's exactly why it is desirable to pack structures if it's a _binary interface_.  32-bit and 64-bit may align things differently depending on circumstance and types used.  

Consider the (possible) case where the stubdom moves to 64-bit while dom0 remains 32-bit, or vice versa.  Assume the same compiler (gcc).  Say one of the structures looked like:

typedef struct {
  uint32_t a;
  uint64_t b;
} uhoh_t;

on your x86-32, sizeof(uhoh_t) == 12.  however, on the other side, sizeof(uhoh_t) == 16 (because x86-64 will shuffle the uint64_t to the 8 byte alignment while x86-32 will do it to 4 byte).

uh oh. thank goodness rian put in a million assertions ;)